File: 58.length-of-last-word/solution.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int last = s.length() - 1;
		for (int i = last; i >= 0; i--)
		{
			if (s[i] != ' ')
			{
				s.erase(i + 1, last - i);
				break;
			}
		}
		last = s.length() - 1;
        for (int i = 0; i <= last; i++)
            if (s[last - i] == ' ')
                return i;
        return s.length();
    }
};

File: 58.length-of-last-word/solution.go
func lengthOfLastWord(s string) int {
	if len(s) == 0 {
		return 0
	}
	last := 0
	max := 0
	for i, c := range s {
		if c == ' ' {
			if i - last > 0 {
				max = i - last
			}
			last = i + 1
		}
	}
	if len(s)- last > 0 {
		max = len(s) - last
	}
	return max
}
